native/src/components/RemoteContent.tsx

```diff
@@ -57,29 +72,67 @@ const RemoteContent = (props: RemoteContentProps): ReactElement | null => {
     }
   }, [onLoad, webViewHeight, content])
 
-  // messages are triggered in renderHtml.ts
-  const onMessage = useCallback((event: WebViewMessageEvent) => {
-    const message = JSON.parse(event.nativeEvent.data)
-    if (message.type === HEIGHT_MESSAGE_TYPE && typeof message.height === 'number') {
-      setWebViewHeight(message.height)
-      return
-    }
+  useFocusEffect(
+    useCallback(() => {
+      appSettings.loadExternalSourcePermissions().then(setExternalSourcePermissions).catch(reportError)
+    }, []),
+  )
 
-    if (message.type === WARNING_MESSAGE_TYPE) {
-      log(message.message, 'warning')
-    } else {
-      const error = new Error(message.message ?? 'Unknown message received from webview')
-      reportError(error)
-      setError(error.message)
+  // Webview content does not reload on opt-in click
+  useEffect(() => {
+    if (reloadSettings) {
+      appSettings.loadExternalSourcePermissions().then(setExternalSourcePermissions).catch(reportError)
+      setReloadSettings(false)
     }
-  }, [])
+  }, [reloadSettings])
```

Comment:
Why did you choose to use a `useEffect` for this instead of just calling a function directly?